### PR TITLE
s390: Extend image handling for patched zKVM upgrades

### PR DESCRIPTION
**Pull request by Soulofdestiny**

### CODE REVIEW

**Comment by Soulofdestiny** [comment]
Verification run:
http://opeth.suse.de/tests/4997

@mnowaksuse @dgutu-suse

File: tests/installation/bootloader_zkvm.pm [hunk 1]
```diff
@@ -76,8 +77,15 @@ sub run() {
...
-        type_string("# copying image...\n");
-        $svirt->add_disk({file => $hdd_path, dev_id => 'a'});    # Copy disk to local storage
+        if (get_var('PATCHED_SYSTEM')) {
+            diag('in patched systems just load the patched image');
+            my $patched_img = "$img_path/$name" . "a.img";
+            $svirt->add_disk({file => $patched_img, dev_id => 'a'});
+        }
+        else {
+            type_string("# copying image...\n");
+            $svirt->add_disk({file => $hdd_path, backingfile => 1, dev_id => 'a'});    # Copy disk to local storage
```
Comment (mnowaksuse) [inline tests/installation/bootloader_zkvm.pm]: > `The key sent before the reconnection, seems to be dropped`

I think this is misleading nowadays. Every image gets copied by the backend.

Comment (okurz) [inline tests/installation/bootloader_zkvm.pm]: I guess we definitely need the 'PATCHED_SYSTEM' branch. Do you argue that we don't need the else branch? but it has been there in before so you are saying it should also be changed additionally or is there a patch where this patch needs to be rebased on?

**Comment by okurz** [comment]
I did a code-walkthrough with @Soulofdestiny together. LGTM but needs verification that default installation still works.

Potentially some "uninitialized" warnings could pop up in the use of `get_var` here. better specify default values.

**Comment by okurz** [comment]
zkvm installation verification: http://lord.arch/tests/6132

and I did not find any warnings so unconditional LGTM by me

**Comment by QingmingSu** [comment]
Seems "S390_ZKVM" is a new variable only for s390x+zkvm. With this new variable, I guess we need create new offline_migration and offline_zdup_migration test suites with suffix specially for s390x architecture with variable S390_ZKVM=1, right?
Can we use combination of some existing variables like "(check_var('ARCH', 's390x') and check_var('BACKEND', 'svirt'))"? if so, we can reuse the same test suites existing for x86_64 architecture, such as: 
  migration_offline_sle12sp1_allpatterns_fullupdate
  migration_offline_sle12sp2-allpatterns_fullupdate   
  migration_offline_sle12sp2_sdk+we+allpatterns_fullupdate
  ......

Too many test suites for offline/online migration tests already exist.

**Comment by Soulofdestiny** [comment]
s390_zKVM is not that new, but yeah it's not used that much
s390_ZKVM=1 is indeed the same as (check_var('ARCH', 's390x') and check_var('BACKEND', 'svirt'))

But we already use the existing testsuites for zkvm migrations:
https://openqa.suse.de/tests/862852
https://openqa.suse.de/tests/862853

**Comment by okurz** [comment]
@Soulofdestiny so can we merge it then?

**Comment by Soulofdestiny** [comment]
@okurz I don't see any issues with merging it, but remember, there is a weekend in between ;)

**Comment by okurz** [comment]
@Soulofdestiny good point. I merged now then, so use the day for monitoring :-)

**Pull request merged**